### PR TITLE
feat(gh-actions) helm package & publish

### PR DESCRIPTION
**Pull request by kengou**
## Description

Implement GitHub Actions to package and push `Helm Charts` into `ghcr.io packages`

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

- Related Issue #13

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help
- [ ] Separate ticket for tests # (issue/pr)

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Added to documentation?

- [ ] 📜 README.md
- [ ] 🤝 Documentation pages updated
- [x] 🙅 no documentation needed
- [ ] (if applicable) generated OpenAPI docs for CRD changes

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by auhlig** [comment]
I thought only the Helm charts in [greenhouse-extensions](https://github.com/cloudoperators/greenhouse-extensions) need to be in the registry, no?

**Comment by kengou** [comment]
> I thought only the Helm charts in [greenhouse-extensions](https://github.com/cloudoperators/greenhouse-extensions) need to be in the registry, no?

I thought in both places Helm charts need to be in the registry.
@IvoGoman What do you think ?

**Comment by IvoGoman** [comment]
IMO we should have the charts for Greenhouse itself and the Extensions in the registry. This way it is no longer necessary to checkout the repo to install Greenhouse into the clusters.

File: .github/workflows/helm-push.yaml [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+          - chartDir: charts/demo
+            chartName: demo
+          - chartDir: charts/greenhouse
+            chartName: greenhouse
+          - chartDir: charts/headscale
+            chartName: headscale
+          - chartDir: charts/idproxy
+            chartName: idproxy
+          - chartDir: charts/logshipping
+            chartName: logshipping
```
Comment (IvoGoman) [inline .github/workflows/helm-push.yaml:35]: This chart got removed

Comment (kengou) [reply .github/workflows/helm-push.yaml]: chart removed

**Review by IvoGoman — APPROVED** [review]

**Pull request merged**